extension/observer/ecsobserver/fetcher.go

```diff
@@ -195,23 +196,25 @@ func (f *taskFetcher) getDiscoverableTasks(ctx context.Context) ([]*ecs.Task, er
 }
 
 // attachTaskDefinition converts ecs.Task into a taskAnnotated to include its ecs.TaskDefinition.
-func (f *taskFetcher) attachTaskDefinition(ctx context.Context, tasks []*ecs.Task) ([]*taskAnnotated, error) {
+func (f *taskFetcher) attachTaskDefinition(ctx context.Context, tasks []ecstypes.Task) ([]*taskAnnotated, error) {
 	svc := f.ecs
 	// key is task definition arn
-	arn2Def := make(map[string]*ecs.TaskDefinition)
+	arn2Def := make(map[string]*ecstypes.TaskDefinition)
 	for _, t := range tasks {
-		arn2Def[aws.StringValue(t.TaskDefinitionArn)] = nil
+		if t.TaskDefinitionArn != nil {
```

Comment:
A couple tests run into null-pointer exceptions if this nil-check is removed and fail after this nil-check is added. I'll need to spend some more time digging into why and how the mock interacts with this, but thought I would share this draft PR in the meantime in case someone else understands this better.